tensorflow/java/src/gen/perl/tftypes-runall.pl

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/perl
+#
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+my $script = $0;
+my $dir = `dirname $script`;
+chomp $dir;
+my $gen = "$dir/..";
+my $tfjavasrc = "$gen/..";
+my $rsrc = "$gen/resources";
+my $root = "$tfjavasrc/main/java";
+my $pkg = "$root/org/tensorflow";
+
+sub locchk {
+    (my $f) = @_;
+    if (! -r $f) {
+        print STDERR "Script tftypes-runall seems to be located in the wrong place (could not find $f)\n";
+        exit 1;
+    }
+}
+&locchk("$gen");
+&locchk("$tfjavasrc/gen");
+&locchk("$dir/tftypes.pl");
+&locchk("$rsrc/tftypes.csv");
+
+system("perl $dir/tftypes.pl -t $rsrc/tftypes.csv $pkg/types");
+# system("perl $dir/tftypes.pl -c $rsrc/tftypes.csv $rsrc/Tensors.java.tmpl > $pkg/op/Tensors.java");
```

Comment:
The value of Tensors (which doesn't really exist in this PR) is that it provides real static checking that Tensor does not. Tensor.create leads to run-time checking. Unfortunately statically checking this requires duplicating all creators for all ranks we want to support, because Java does not support rank polymorphism. Note that Constant and Variable ops will want to work similarly.